src/AMAMM.sol

```diff
@@ -92,98 +85,81 @@ contract AMAMM is IAmAmm {
         // - deposit needs to be a multiple of rent
         // - payload needs to be valid
         if (
-            bidder == address(0) || rent <= poolEpochManager[id][_epoch].rent.mulWad(MIN_BID_MULTIPLIER(id))
-                || deposit < rent * K(id) || deposit % rent != 0 || !_payloadIsValid(id, payload)
+            rent <= poolEpochManager[id][_epoch].rent.mulWad(MIN_BID_MULTIPLIER(id)) || deposit < rent * K(id)
+                || deposit % rent != 0 || !_payloadIsValid(id, payload)
         ) {
             revert AmAmm__InvalidBid();
         }
 
         // Check if the bid already exists
-        Bid memory existingBid = poolEpochBids[id][_epoch][bidder];
         Bid memory newBid = Bid({bidder: msgSender, payload: payload, rent: rent, deposit: deposit});
 
-        if (existingBid.bidder != address(0)) {
-            // If the bid exists, update the deposit
-            existingBid.deposit += deposit;
-        } else {
-            // If the bid does not exist, create a new one
-            poolEpochBids[id][_epoch][bidder] = newBid;
-        }
-
-        //Update poolEpochManager if deposit is largest
         if (poolEpochManager[id][_epoch].deposit < deposit) {
-            poolEpochManager[id][_epoch] = newBid;
+            _userRefunds[poolEpochManager[id][_epoch].bidder] += poolEpochManager[id][_epoch].deposit;
+            poolEpochManager[id][_epoch] = Bid({bidder: msgSender, payload: payload, rent: rent, deposit: deposit});
+            _userDeposits[msgSender] += deposit;
```

Comment:
Deposit should be before the validation check so that it can has enough balance to pay for the rent. And I reckon it should be a separate deposit function as it also need to handle the bitToken transfer.